CONTRIBUTING.md

```diff
@@ -0,0 +1,54 @@
+# MIOpen Contributing Guide
+To ensure the quality of the MIOpen code base, the MIOpen team has 
+established a code review process to inform developers of the steps 
+that are required to shepherd a change-set into the repository.
+
+## Creating a Pull Request
+No changes are allowed to be directly committed to the develop 
+branch of the MIOpen repository. All authors are required to 
+develop their change sets on a separate branch and then create 
+a pull request (PR) to merge their changes into the develop branch.
+
+Once a PR has been created, a developer must choose two reviewers 
+to review the changes made. The first reviewer should be a 
+technical expert in the portion of the library that the changes 
+are being made in. You can find a list of these experts in 
+[MIOpen Issue #789](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/789)
+. The second reviewer should be a peer reviewer. This reviewer 
+can be any other MIOpen developer.
+
+## Responsibility of Author
+The author of a PR is responsible for:
+ * Writing clear, well documented code
+ * Meeting expectations of code quality
+ * Verifying that the changes do not break current functionality
+ * Writing tests to ensure code coverage
+ * Report on the impact to performance
+
+## Responsibility of the Reviewer
```

Comment:
A suggestion is to make the reviewer's task clear and simple as a checklist:
1. Has the PR passed necessary CI? Check
2. Has the PR been rebased and squashed to a limited number of meaningful commits only? e.g. < 5
3. Does the PR only include a reviewable amount of changes? Or it is a consolidation of already reviewed small batches? e.g. break it into smaller testable and reviewable tasks instead of a huge chunk at once
4. Sufficient documentation and easy to read/understand/test/maintain, related docs already in pace? e.g. revise or add to https://rocmsoftwareplatform.github.io/MIOpen/doc/html/
5. For bugfixes and new features, new regression test created and included in CI, or some other holistic test pipeline?
6. Is every PR associated with a ticket or issue number for tracking purposes?

The default action for reviewers: deny PR with reasons (I will leave this one for discussion :))

Reply:
> Has the PR been rebased and squashed to a limited number of meaningful commits only? e.g. < 5

Usually we do not impose any restrictions on developer commits, because on merging, we squash all developer commits anyway.

However it is obvious that a well-organized sequence of small commits, each of which is designed to make one specific feature or fix (and ideally should be able to pass CI testing), is very good, as it helps to review the code. Well-written commit descriptions are also important. It may worth mentioning this in the contributing guide.